server/src/main/java/org/opensearch/action/search/SearchTask.java

```diff
@@ -80,7 +80,11 @@ public SearchTask(
 
     @Override
     public final String getDescription() {
-        return descriptionSupplier.get();
+        try {
```

Comment:
@dblock so I've check the codebase and it seems like we have at least 4 consolidated places where `UnsupportedOperationException` is thrown all over the place:
 - GeoJson (JSON serialization)
 - GeometryIO (native stream serialization)
 - GeoShapeIndexer (indexer)
 - LegacyGeoShapeQueryProcessor / VectorGeoPointShapeQueryProcessor / LuceneGeometryCollector

 It seems like `LinearRing` is special case and should not be used as a top level geometry primitive, looking a bit more on how we could fix that.